shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -887,6 +887,26 @@ public static String validateUserStreamName(String name) {
         return name;
     }
 
+    /**
+     * Validates a user created stream.
+     *
+     * @param name User supplied stream name to validate.
+     * @return True, if its a user created stream.
+     */
+    public static boolean isUserStreamSegment(String name) {
```

Comment:
The semantics of this should be made more clear. Think about how this works with transactions, transients, reader groups, index segments, and table segments.

Reply:
I have added check for others segments. This will be used to validate user created stream segments only. For other segments like index, transaction etc it will return false.